projects/core/src/site-context/config/config-loader/site-context-config-initializer.spec.ts

```diff
@@ -0,0 +1,141 @@
+    uid: 'test',
+    urlPatterns: [''],
+    baseStore: mockBaseStore,
+    urlEncodingAttributes: ['language', 'currency'],
+    theme: 'test-theme',
+  },
+];
+
+class MockBaseSiteService {
+  getAll(): Observable<BaseSite> {
+    return of({});
+  }
+}
+
+// TODO(#11515): remove it in 4.0
+class MockFeatureConfigService {
+  isLevel() {
+    return true;
+  }
+}
+
+class MockSiteContextConfig {
+  context = {};
+}
+
+describe(`SiteContextConfigInitializer`, () => {
+  let initializer: SiteContextConfigInitializer;
+  let baseSiteService: BaseSiteService;
+  let config: SiteContextConfig;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: BaseSiteService, useClass: MockBaseSiteService },
+        { provide: WindowRef, useClass: MockWindowRef },
+        {
+          provide: JavaRegExpConverter,
+          useValue: {
+            toJsRegExp: jasmine.createSpy().and.callFake((x) => new RegExp(x)),
+          },
+        },
+        { provide: FeatureConfigService, useClass: MockFeatureConfigService },
+        { provide: SiteContextConfig, useClass: MockSiteContextConfig },
+      ],
+    });
+
+    initializer = TestBed.inject(SiteContextConfigInitializer);
+    baseSiteService = TestBed.inject(BaseSiteService);
+    config = TestBed.inject(SiteContextConfig);
+  });
+
+  describe(`resolveConfig - context was already configured statically`, () => {
+    it(`should return empty object`, async () => {
+      config.context = {
+        baseSite: ['electronics'],
+      };
+      const result = await initializer.configFactory();
+      expect(result).toEqual({});
+    });
+  });
+
+  describe(`resolveConfig - context was not already configured statically`, () => {
+    it(`should throw error when the base sites loaded are undefined`, async () => {
+      spyOn(baseSiteService, 'getAll').and.returnValue(of(undefined));
+      let message = false;
+      try {
+        await initializer.configFactory();
+      } catch (e) {
+        message = e.message;
+      }
+      expect(message).toBeTruthy();
+    });
+
+    it(`should throw error when the base sites loaded is an empty array`, async () => {
+      spyOn(baseSiteService, 'getAll').and.returnValue(of([]));
+      let message = false;
+      try {
+        await initializer.configFactory();
+      } catch (e) {
+        message = e.message;
+      }
+      expect(message).toBeTruthy();
+    });
+
+    it(`should throw error when no url pattern of any base site matches the current url`, async () => {
+      initializer['isCurrentBaseSite'] = () => false;
+      spyOn(baseSiteService, 'getAll').and.returnValue(of(mockBaseSites));
+
+      let message = false;
+      try {
+        await initializer.configFactory();
+      } catch (e) {
+        message = e.message;
+      }
+      expect(message).toBeTruthy();
+    });
+
+    it(`should return config based on loaded sites data`, async () => {
+      initializer['isCurrentBaseSite'] = () => true;
```

Comment:
We're only mocking isCurrentBaseSite. In a separate test it would be good to also test it's internal behavior too (i.e. using java regex converter)